autoload/ale/sign.vim

```diff
@@ -67,10 +67,14 @@ endfunction
 " Given a loclist, combine the loclist into a list of signs such that only
 " one sign appears per line. Error lines will take precedence.
 " The loclist will have been previously sorted.
-function! ale#sign#CombineSigns(loclist) abort
+function! ale#sign#CombineSigns(buffer, loclist) abort
     let l:signlist = []
 
     for l:obj in a:loclist
+        if has_key(l:obj, 'bufnr') && l:obj.bufnr >= 0 && l:obj.bufnr != a:buffer
```

Comment:
@w0rp I needed to prevent signs from other files from appearing in the primary buffer. what do you think of this?

It may be able to be simplified to just `if l:obj.bufnr != a:buffer`, but I wasn't sure if `bufnr` is always set.